packages/assistify-ai/server/methods/SmartiWidgetBackend.js

```diff
@@ -164,6 +164,11 @@ Meteor.methods({
 		};
 	},
 
+	closeConversation(message, roomId, user) {
+		RocketChat.models.Messages.createWithTypeRoomIdMessageAndUser('request_closed', roomId, ` ${ message }`, user);
+		RocketChat.archiveRoom(roomId);
```

Comment:
The function on the model does not perform authorization checks. The method should be used instead: `Meteor.call('archiveRoom', roomId)` should be used.
Currently, all members of the channel can close the room - though they were not allowed to archive it

Reply:
That is exactly the problem. Every member of a thread should be able to close the the thread (archive it). Therefore I can not use `Meteor.call('archiveRoom', roomId)`.
I could add a check for `user exists` or `room exists`. But in general, I think it is worse to have people who are not able to close the room but want to, then having someone archiving the room when he shouldn't. Especially since archiving is not irreversible.

Reply:
The `ChatRooms`-collection on the frontend is a cached collection. I'm not 100% sure that the archived-property is part of that though.
Still, accessing the database asynchronously, should not be an issue.